### PR TITLE
Skit 563 fix

### PR DESCRIPTION
**Pull request by Ryanmtate**
https://github.com/spruceid/oid4vp-rs/pull/19 merged into the wrong branch. 

Pushing this branch to fix the incorrect merge.

### CODE REVIEW

**Comment by Ryanmtate** [comment]
This PR also rebased on top of main.

File: tests/presentation-definition/iso.org.18013.5.1.mdl.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        "fields": [
+          {
+            "path": ["$.type"],
+            "filter": {
+              "type": "array",
+              "contains": {
+                "const": "iso.org.18013.5.1.mDL"
+              }
+            }
+          },
```
Comment (cobward) [inline tests/presentation-definition/iso.org.18013.5.1.mdl.json]: This is not a valid mDL presentation definition, so I don't think we should add this to the repo as a test.

File: tests/presentation-definition/multi-credential-array.json [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                "const": "EducationalCredential"
+              }
+            }
+          },
+          {
+            "path": ["$.credentialSubject.firstName", "$.vc.credentialSubject.firstName"],
+            "purpose": "The credential must contain the holder's first name",
+            "filter": {
+              "type": "string",
+              "minLength": 1
```
Comment (cobward) [inline tests/presentation-definition/multi-credential-array.json]: NABD but I don't think the `minLength: 1` filter is necessary
```suggestion
              "type": "string"
```

File: tests/presentation-definition/multi-credential-array.json [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      "alg": ["EdDSA", "ES256K"]
+    },
+    "jwt_vc": {
+      "alg": ["EdDSA", "ES256K"]
+    },
+    "ldp_vc": {
+      "proof_type": ["Ed25519Signature2018", "EcdsaSecp256k1Signature2019"]
+    },
+    "ldp_vp": {
+      "proof_type": ["Ed25519Signature2018", "EcdsaSecp256k1Signature2019"]
```
Comment (cobward) [inline tests/presentation-definition/multi-credential-array.json]: NABD: we rarely use secp256k1, secp256r1 is our default ECDSA curve
```suggestion
      "alg": ["EdDSA", "ES256"]
    },
    "jwt_vc": {
      "alg": ["EdDSA", "ES256"]
    },
    "ldp_vc": {
      "proof_type": ["Ed25519Signature2018", "EcdsaSecp256r1Signature2019"]
    },
    "ldp_vp": {
      "proof_type": ["Ed25519Signature2018", "EcdsaSecp256r1Signature2019"]
```

**Review by cobward — APPROVED** [review]

**Pull request merged**